fluss-server/src/test/java/com/alibaba/fluss/server/kv/rocksdb/RocksDBMetricsManagerTest.java

```diff
@@ -0,0 +1,423 @@
+        manager.shutdown();
+        assertThat(manager.isShutdown()).isTrue();
+    }
+
+    @Test
+    void testConcurrentRegistrationAndUnregistration() throws Exception {
+        RocksDBMetricsManager manager = RocksDBMetricsManager.getInstance();
+
+        int threadCount = 10;
+        CountDownLatch latch = new CountDownLatch(threadCount);
+        ExecutorService executor = Executors.newFixedThreadPool(threadCount);
+        AtomicInteger collectorId = new AtomicInteger(0);
+
+        for (int i = 0; i < threadCount; i++) {
+            executor.submit(
+                    () -> {
+                        try {
+                            for (int j = 0; j < 10; j++) {
+                                RocksDBMetricsCollector collector =
+                                        createMockCollector(
+                                                "test_db",
+                                                "test_table",
+                                                collectorId.getAndIncrement());
+                                manager.registerCollector(collector);
+                                collectors.add(collector);
+
+                                // Immediately unregister some collectors
+                                if (j % 2 == 0) {
+                                    manager.unregisterCollector(collector);
+                                }
+                            }
+                        } finally {
+                            latch.countDown();
+                        }
+                    });
+        }
+
+        latch.await(10, TimeUnit.SECONDS);
+        executor.shutdown();
+
+        // Should handle concurrent access gracefully
+        assertThat(manager.getCollectorCount()).isGreaterThanOrEqualTo(0);
+    }
+
+    @Test
+    void testEmptyCollectorsDoesNotCauseIssues() throws Exception {
+        RocksDBMetricsManager manager = RocksDBMetricsManager.getInstance();
+
+        // Start with no collectors
+        assertThat(manager.getCollectorCount()).isEqualTo(0);
+
+        // Wait a bit to ensure no exceptions are thrown
+        Thread.sleep(1000);
+
+        // Should still be not running
+        assertThat(manager.isRunning()).isFalse();
+    }
+
+    @Test
+    void testCreateReporterKey() {
+        RocksDBMetricsCollector collector1 = createMockCollector("db1", "table1", 0);
+        RocksDBMetricsCollector collector2 = createMockCollector("db1", "table1", 1);
+        RocksDBMetricsCollector collector3 = createMockCollector("db2", "table1", 0);
+
+        // Different buckets should have different keys
+        assertThat(collector1.getTableBucket().getBucket())
+                .isNotEqualTo(collector2.getTableBucket().getBucket());
+
+        // Different databases should have different keys
+        assertThat(collector1.getTablePath().getDatabaseName())
+                .isNotEqualTo(collector3.getTablePath().getDatabaseName());
```

Comment:
I don't see it asserts anything. If we want to assert the reporter key is different, I think we need to assert `manager.getCollectorCount() = 3` here?